site/content/features/stringtemplates.md

```diff
@@ -0,0 +1,466 @@
+import static java.lang.StringTemplate.RAW;
+import java.util.List;
+
+public class StringTemplateDemo {
+   public static void main(String[] args) {
+      String name = "Fred";
+      int age = 42;
+      StringTemplate template = RAW."Hello, \{name}! Next year, you'll be \{age + 1}.";
+      System.out.println(template.fragments());
+      System.out.println(template.values());
+
+      // Two interpolate methods: one static, one instance
+      System.out.println(StringTemplate.interpolate(
+         template.fragments().stream().map(String::toUpperCase).toList(), template.values()));
+      System.out.println(StringTemplate.of(
+         template.fragments().stream().map(String::toUpperCase).toList(),
+            template.values()).interpolate());
+
+      // Demo of the of and combine methods.
+      // For each entry in System.getProperties(), one StringTemplate is created.
+      // They are all combined and then processed with the STR processor
+      template = StringTemplate.combine(
+         System.getProperties()
+            .entrySet()
+            .stream()
+            .map(e -> StringTemplate.of(List.of(e.getKey() + ": ", "\n"),
+               List.of(e.getValue())))
+            .toList());
+      System.out.println(template.process(STR));
+   }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+## Writing Your Own Template Processor
+
+The `format` method of a template processor turns `StringTemplate` instances into objects. Since `StringTemplate.Processor` is a functional interface, you can construct an instance from a lambda expression. Here is a simple example that places the values in boxes:
+
+{{< sandbox version=java19 preview="true" mainclass="MyFirstTemplateProcessor" >}}{{< sandboxsource "MyFirstTemplateProcessor.java" >}}
+public class MyFirstTemplateProcessor {
+   public static StringTemplate.Processor<String, RuntimeExpression> BOX =
+      (StringTemplate template) -> {
+         StringBuilder result = new StringBuilder();
+         for (int i = 0; i < template.fragments.size(); i++) {
+            if (i > 0) {
+               result.append('[');
+               result.append("" + template.values.get(i));
+               result.append(']'); 
+            }
+            result.append(fragments.get(i));
+         }
+         return result.toString();
+      };
+
+   public static void main(String[] args) {
+      System.out.println(BOX."Hello, \{name}! Next year, you'll be \{age + 1}.");
+   }
+}
+
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+`StringTemplate.Processor` is a generic class with two type parameters: the result type and the type of the exception that the processor can throw. 
+
+Here, the values are transformed and then interpolated with the fragments. The `interpolate` method is helpful in this common situation:
+
+```
+public static StringTemplate.Processor<String, RuntimeException> BOX =
+   (StringTemplate template) -> StringTemplate.interpolate(template.fragments(),
```

Comment:
Could be simpiified, also i the next snippet:

    template -> StringTemplate.interpolate(template.fragments(),